src/nrniv/nrncore_write.cpp

```diff
@@ -238,17 +238,26 @@ static void part2(const char* path) {
   }
 
   // filename data might have to be collected at hoc level since
-  // pc.nrnbbcore_write might be called
+  // pc.nrncore_write might be called
   // many times per rank since model may be built as series of submodels.
-  if (ifarg(2)) {
+  if (ifarg(2) && hoc_is_object_arg(2) && is_vector_arg(2)) {
```

Comment:
isn't `is_vector_arg(2)` sufficient?

Reply:
Sadly, the is_vector_arg assumes the arg is at least  an Object.